### PR TITLE
WIP: Add a page ToC to docs.

### LINKED ISSUE #611: Add a "On this page" secondary nav for the documentation
This is inspired by the likes of:
* https://docs.cypress.io/guides/overview/why-cypress
* https://tailwindcss.com/docs/installation

https://www.npmjs.com/package/rehype-toc looks promising.

### PR DESCRIPTION
**Pull request by mikenikles**
Close #611.

A few things we need to look into:
- [ ] Reduce the spacing between the left-side nav and the docs content
- [ ] Increase the content width (according to how much we reduce the left-side nav)
- [ ] Style the page table of contents
- [ ] Hide the page table of contents on the blog pages (or not 🤔?)
- [ ] Scroll behaviour:
  - [ ] Left-side nav: Fixed
  - [ ] Docs content: Scroll vertically
  - [ ] Right-side page-level nav: Fixed

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/613"></a>

### CODE REVIEW

**Comment by nisarhassan12** [comment]
@mikenikles Sorry I just forced pushed here somehow I messed up the `package
-lock.json` file while merging the branch with the changes from the latest main. To get rid of the bad merged I nuked that commit and forced pushed to get back to the initial stage and then did the merge carefully.

**Comment by nisarhassan12** [comment]
> - [ ] Left-side nav: Fixed
> - [ ] Docs content: Scroll vertically
> - [ ] Right-side page-level nav: Fixed

@mikenikles With the above implemented how do we envision the footer do we want to use a slightly different footer with fewer contents just like https://docs.cypress.io/guides/overview/why-cypress is doing it i.e on the footer on their site they have a lot more content.

File: src/components/main-nav/index.svelte [hunk 1]
```diff
@@ -32,11 +35,28 @@
...
+    }
+  });
 </script>
 
+<style lang="scss">
+  @media (min-width: 769px) {
+    .is-on-docs {
+      left: 50%;
+      transform: translateX(-50%);
+      @apply fixed bg-sand-light z-10;
```
Comment (nisarhassan12) [inline src/components/main-nav/index.svelte:51]: I'll look into changing these to tailwind classes later tonight.

**Comment by mikenikles** [comment]
Thanks Nisar for looking into that.

It looks quite complex with some booleans (e.g. `isOnDocs` and `isHidden`) as well as numerous pixel values to define widths and new media queries etc. Let’s put this on hold for now. This takes a lot more time than I expected and also looks more complicated than I hoped. When I looked at the Tailwind docs initially, it looked like a 3-column layout with the one on the very right hidden on smaller devices - all achievable with Tailwind classes.

I need to figure out why ours turned out so complex and we can then discuss how we want to proceed.

**Comment by mikenikles** [comment]
Closing in favour of #1296

**Pull request closed without merging**